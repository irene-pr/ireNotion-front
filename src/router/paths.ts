const paths = {
  home: "/",
  login: "/login",
  register: "/register",
  userBoard: "/userBoard",
};

export default paths;
