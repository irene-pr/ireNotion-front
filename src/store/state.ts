const state = {
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
};

export default state;
