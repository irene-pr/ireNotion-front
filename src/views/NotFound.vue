<template>
  <main class="not-found-page">
    <h1 class="not-found-page__header">
      <span class="span-1">404</span> <span class="span-2">Not </span>
      <span class="span-3">found</span>
    </h1>
  </main>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "NotFound",
});
</script>

<style scoped lang="scss">
@import "@/styles/_variables";

.not-found-page {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url("../assets/background-3.webp");
  background-repeat: no-repeat;
  background-size: cover;
  &__header {
    color: $theme-light-color;
    font-size: 50px;
    .span-1 {
      color: #f5bc76;
      letter-spacing: 0.1em;
      text-shadow: 6px 3px rgb(93, 1, 116);
    }
    .span-2 {
      font-style: italic;
      font-weight: 100;
    }
    .span-3 {
      color: #333fe4;
      font-family: $Lato;
      font-weight: 900;

      //-webkit-text-stroke: 1px azure;
    }
    @media only screen and (min-width: 480px) {
      font-size: 65px;
    }
    @media only screen and (min-width: 720px) {
      font-size: 80px;
    }
  }
}
</style>
