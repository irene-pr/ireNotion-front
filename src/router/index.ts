import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";
import paths from "@/router/paths";
import Home from "../views/Home.vue";
import Login from "../views/Login.vue";
import Register from "../views/Register.vue";
import BoardPage from "../views/BoardPage.vue";

const routes: Array<RouteRecordRaw> = [
  {
    path: paths.home,
    name: "Home",
    component: Home,
  },
  {
    path: paths.login,
    name: "Login",
    component: Login,
  },
  {
    path: paths.register,
    name: "Register",
    component: Register,
  },
  {
    path: paths.userBoard,
    name: "BoardPage",
    component: BoardPage,
  },
];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
