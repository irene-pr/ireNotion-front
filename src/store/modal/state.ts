import { IModalState } from "@/types/store";

const state = (): IModalState => ({
  isBoardEditModal: false,
  isUpdateNoteModal: false,
  idForModal: "",
});

export default state;
