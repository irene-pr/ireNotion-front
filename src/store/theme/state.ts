import { IThemeState } from "@/types/store";

const state = (): IThemeState => ({
  themeHeaders: "night-mode",
  themeSurfaces: "day-mode",
});

export default state;
