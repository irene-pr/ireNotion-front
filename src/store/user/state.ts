import { IUserState } from "@/types/store";

const state = (): IUserState => ({
  isLoggedIn: false,
  userContent: {
    name: "",
    boards: [],
  },
  userData: {
    userId: "",
    username: "",
  },
  isLoading: false,
});

export default state;
