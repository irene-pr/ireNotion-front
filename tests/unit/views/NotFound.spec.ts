import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import router from "@/router";
import state from "@/store/state";
import NotFound from "../../../src/views/NotFound.vue";

describe("Given a Home view", () => {
  const store = createStore({
    state() {
      return state;
    },
  });

  describe("When it is rendered", () => {
    test("Then it renders", () => {
      mount(NotFound, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });
    });
    test("Then it has a heading", () => {
      const wrapper = mount(NotFound, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const heading = wrapper.find("h1");

      expect(heading.exists).toBeTruthy();
    });
    test("Then the heading has text '404 Not found'", () => {
      const wrapper = mount(NotFound, {
        global: {
          plugins: [router, store],
        },
        stubs: ["router-view"],
      });

      const heading = wrapper.find("h1");

      expect(heading.text()).toBe("404 Not found");
    });
  });
});
