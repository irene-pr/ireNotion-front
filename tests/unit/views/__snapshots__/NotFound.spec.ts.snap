// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Home view When it is rendered Then the it matches the snapshot 1`] = `
<main
  class="not-found-page"
>
  <a
    class=""
    href="/"
  >
    <h1
      class="not-found-page__header"
    >
      <span
        class="span-1"
      >
        404
      </span>
       
      <span
        class="span-2"
      >
        Not 
      </span>
      <span
        class="span-3"
      >
        found
      </span>
    </h1>
  </a>
</main>
`;
